nfv_ovs_dpdk.md

```diff
@@ -0,0 +1,262 @@
+# NFV OVS DPDK
+
+This document describes how to deploy EDPM nodes with OVS DPDK.
+
+## Procedure Overview
+
+1. Configure kernel args and tuned parameters
+2. Configure OVS DPDK parameters
+3. Configure the networks with DPDK interfaces of the EDPM nodes
+2. Configure Nova and Neutron parameters.
+
+In order to complete the above procedure, the `services` list of the
+`OpenStackDataPlaneNodeSet` CR needs to be edited.
+
+## OpenStackDataPlaneNodeSet services list
+
+EDPM nodes can be configured by creating an
+`OpenStackDataPlaneNodeSet` CR which the
+[dataplane-operator](https://openstack-k8s-operators.github.io/dataplane-operator)
+will reconcile when an `OpenStackDataPlaneDeployment` CR is created.
+These types of CRs have a `services` list like the following:
+
+```yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneNodeSet
+spec:
+  ...
+  services:
+    - bootstrap
+    - download-cache
+    - reboot-os
+    - configure-ovs-dpdk
+    - configure-network
+    - validate-network
+    - install-os
+    - configure-os
+    - ssh-known-hosts
+    - run-os
+    - install-certs
+    - ovn
+    - neutron-ovn
+    - neutron-metadata
+    - libvirt
+    - nova-custom-ovsdpdk
+```
+Only the services which are on the list will be configured.
+
+## Configure the EDPM ansible variables of the EDPM nodes
+
+Following are the list of EDPM ansible variables which need to be
+provided for deploying with OVS DPDK support.
+
+edpm_ovs_dpdk_pmd_core_list: List of Logical CPUs to be allocated for Poll Mode Driver 
+
+edpm_ovs_dpdk_socket_memory: Socket memory list per NUMA node
+
+edpm_ovs_dpdk_memory_channels: Number of memory channel per socket
+
+edpm_ovn_bridge_mappings: List of bridge and dpdk ports mappings
+
+edpm_kernel_args: Configures kernel boot parameters
+
+edpm_tuned_profile: tuned profile configuration
+
+edpm_tuned_isolated_cores: List of isolated cores
+
+edpm_nova_libvirt_qemu_group: Configures qemu group for nova libvirt
+
+
+Example:
+dpm_ovs_dpdk_pmd_core_list: "1,13,2,14,3,15"
+edpm_ovs_dpdk_socket_memory: "4096"
+edpm_ovs_dpdk_memory_channels: "4"
+edpm_ovn_bridge_mappings: ['dpdk2:br-link2','dpdk1:br-link1']
+edpm_kernel_args: "default_hugepagesz=1GB hugepagesz=1G hugepages=64 iommu=pt intel_iommu=on tsx=off isolcpus=2-11,14-23"
+edpm_nova_libvirt_qemu_group: "hugetlbfs"
+edpm_tuned_profile: "cpu-partitioning"
+edpm_tuned_isolated_cores: "2-11,14-23"
```

Comment:
Maybe put this whole block in triple-ticks

Reply:
done